ENH: Add _cmd_prefix class variable to CommandLine

effigies opened the pull request:
This PR adds a `_cmd_prefix` which is a string that is prepended (with no space) to the command. This will permit users to add installation-specific prefixes, whether they be installation directories that are not in the path or idiosyncratic names that have been chosen by a package manager.

The idea is that users could do something like:

```Python
import nipype
nipype.interfaces.ants.AntsCommand._cmd_prefix = '/opt/ants/'
```

or

```Python
import nipype
nipype.interfaces.fsl.FSLCommand._cmd_prefix = 'fsl_'
```

Fixes #1415.

cc @satra @TheChymera

effigies commented:
@satra If you want to have a look at this to make sure it's what you were thinking, that'd be great.

satra commented on nipype/interfaces/afni/base.py:
```diff
@@ -294,13 +294,10 @@ class AFNIPythonCommandInputSpec(CommandLineInputSpec):
-        else:
-            return super(AFNIPythonCommand, self).cmd
+        orig_cmd = super(AFNIPythonCommand, self).cmd
+        found = spawn.find_executable(orig_cmd)
```
i think `find_executable` may need to be updated unless we mandate that the prefix has to be a path prefix returning a fully formed path. 

(at least in terms of our present docs).

satra commented:
in general this looks good.

effigies commented:
@satra Testing found what I think you were thinking of, which was a `which` function. Can you have a quick look at this latest commit?

effigies commented:
@satra I opted to go with [`shlex.split(prefix + cmd)`](https://docs.python.org/3/library/shlex.html#shlex.split), which should be more reliable than any parsing strategy we choose.

effigies commented:
This is ready, as far as I'm concerned.

effigies commented:
@satra This is the only thing left to review for 1.0.
